simpleviewinc/keg-hub
# Linear gradient for web

**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-373)

`[semver-status]: minor`

## Context

* we ran into some issues building the tap because of the new added dependencies `react-native-linear-gradient` && `react-native-web-linear-gradient`
* our app was trying to build with the `react-native` variation instead of `react-native-web`. this looks like an issue with our `keg-components` rollup because none of the build file uses the `react-native-web` variance
* We opt in to just port over the `react-native-web-linear-gradient` code from [here](https://github.com/react-native-web-community/react-native-web-linear-gradient) to keg-components since it's simple enough

## Goal

* Remove the dependency of `react-native-web-linear-gradient` and add the code to keg-components
* Update `TextToggle` to use `kegLinearGradient` instead of the `react-native-web` version. 
      * Native version will still use `react-native-linear-gradient` though
* for native apps, still use `react-native-linear-gradient`
* Fix the rollup and webconfig so it resolves aliases properly

## Updates

* `repos/keg-components/configs/aliases.json`
   * added alias for kegLinearGradient

* Remove `react-native-web-linear-gradient` deps and reference from all the places

* `repos/keg-components/src/components/linearGradient/linearGradient.js`
   * copy pasted logic  identical to the one from their [github](https://github.com/react-native-web-community/react-native-web-linear-gradient/blob/master/src/index.js)
   * minor updates so it's a functional component and we pass in a custom className
* `repos/keg-components/src/components/linearGradient/linearGradient.native.js`
   * on native use `react-native-linear-gradient` still

* `repos/keg-components/src/components/textToggle/textToggle.js`
   * update `TextToggle` to use our version of LinearGradient
## Testing

* run `keg dpg run package=docker.pkg.github.com/simpleviewinc/keg-packages/keg-components:linear-gradient`
* go to http://local.kegdev.xyz/?path=/docs/interactions-texttoggle--basic
* verify that you see the gradient
* inspect the gradient element
* verify you see the classname `keg-linear-gradient`


## build
* pull the branch
* navigate to keg-components
* run `yarn install` > `yarn build`
* verify that the build is successfful without any warnings or error
* verify that `/build/esm/*.native.js` contains an import from `react-native-linear-gradient`
* verify that `/build/esm/*.js` does NOT contain this import but instead has a function called `LinearGradient`
